mscore/timeline.cpp

```diff
@@ -0,0 +1,1521 @@
+                                                         text_width);
+                        text->setPlainText(part_name);
+                        text->setX(pos);
+                        text->setY(height * 3 + verticalScrollBar()->value());
+                        QGraphicsRectItem* rehearse = new QGraphicsRectItem(pos,
+                                                                            height * 3 + verticalScrollBar()->value(),
+                                                                            text->boundingRect().width(),
+                                                                            height);
+                        rehearse->setData(0, QVariant::fromValue<int>(-1));
+                        rehearse->setData(1, QVariant::fromValue<ElementType>(ElementType::REHEARSAL_MARK));
+                        rehearse->setData(2, QVariant::fromValue<void*>(seg->measure()));
+                        rehearse->setData(3, QVariant::fromValue<bool>(true));
+                        rehearse->setData(4, QVariant::fromValue<void*>(e));
+
+                        text->setData(0, QVariant::fromValue<int>(-1));
+                        text->setData(1, QVariant::fromValue<ElementType>(ElementType::REHEARSAL_MARK));
+                        text->setData(2, QVariant::fromValue<void*>(seg->measure()));
+                        text->setData(3, QVariant::fromValue<bool>(true));
+                        text->setData(4, QVariant::fromValue<void*>(e));
+
+                        rehearse->setPen(QPen(Qt::black));
+                        rehearse->setBrush(QBrush(QColor(150,150,150)));
+                        scene()->addItem(rehearse);
+                        scene()->addItem(text);
+
+                        std::pair<QGraphicsItem*, int> rehearse1(rehearse, 3);
+                        std::pair<QGraphicsItem*, int> rehearse2(text, 3);
+                        meta_rows.push_back(rehearse1);
+                        meta_rows.push_back(rehearse2);
+                        }
+                  }
+            }
+
+      //Draw measure numbers
+      QGraphicsRectItem* measure_box = new QGraphicsRectItem(0,
+                                                             height * 4 + verticalScrollBar()->value(),
+                                                             length,
+                                                             height);
+      measure_box->setBrush(QBrush(QColor(211,211,211)));
+      measure_box->setPen(QPen(QColor(150, 150, 150)));
+      measure_box->setData(0, QVariant::fromValue<int>(-1));
+      scene()->addItem(measure_box);
+
+      std::pair<QGraphicsItem*, int> measure_p(measure_box, 4);
+      meta_rows.push_back(measure_p);
+
+      for (int c = 0; c < cols; c++) {
+            if ((c + 1) % 5 == 0 || c == 0) {
+                  //Make sure to center measure numbers and decrease text size
+
+                  QGraphicsTextItem* num = new QGraphicsTextItem(QString::number(c + 1));
```

Comment:
It's debatable, but I think it would be better to find the right font size for the max number of digit and stick with it for all measure numbers. It looks a bit weird otherwise.
<img width="428" alt="capture d ecran 2017-07-06 23 50 35" src="https://user-images.githubusercontent.com/234271/27934404-ebf5cc1a-62a5-11e7-8c02-80e355b9989a.png">